src/Get.tsx

```diff
@@ -277,7 +287,7 @@ function Get<TData = any, TError = any>(props: GetProps<TData, TError>) {
     <RestfulReactConsumer>
       {contextProps => (
         <RestfulReactProvider {...contextProps} parentPath={composePath(contextProps.parentPath, props.path)}>
-          <ContextlessGet {...contextProps} {...props} />
+          <ContextlessGet {...contextProps} {...props} __internal_hasExplicitBase={Boolean(props.base)} />
```

Comment:
Is this introduced to keep track of what was initially passed in, anticipating an override by `defaultProps`? If so, I think it'd be clearer to remove `defaultProps` and implement defaulting logic manually inside.

Reply:
> Is __internal_hasExplicitBase introduced to keep track of what base prop was initially passed in, anticipating an override by defaultProps?

The motivation behind this prop was described in the [type definition](https://github.com/contiamo/restful-react/pull/88/files#diff-63d62a288bee27bda4e83b106d69c46dR62). 🙂 

Essentially, [_everything_ has a `base` prop implicitly via context](https://github.com/contiamo/restful-react/blob/master/src/Get.tsx#L279), and paths compose as you'd expect. 

💩 really hits the fan when a `base` is passed explicitly because the `parentPath` is still appended to the `base`, as outlined in the description of this PR. This is unexpected. It certainly was unexpected in Labs. 

We need some way of knowing "hey, I have an explicit base reset" in order to [tell restful-react that it should disregard `parentPath`](https://github.com/contiamo/restful-react/blob/fix/base-propagation/src/Get.tsx#L223-L227), and so this _internal, non-public-API prop_ was introduced.

Reply:
Sweet, it makes sense now, thanks for clearing it up!